jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostSparkPipelinePersistence.scala

```diff
@@ -48,10 +59,14 @@ class XGBoostSparkPipelinePersistence extends FunSuite
     val same2Model = XGBoostModel.load("test2xgbPipe")
 
     val memoryPredictions = predModel.transform(vectorAssembler.transform(df))
+    memoryPredictions.show
     val loadedPredictions = same2Model.transform(vectorAssembler.transform(df))
+    loadedPredictions.show
+    memoryPredictions.printSchema()
+    loadedPredictions.printSchema()
     // this doesn't work -> will need to compare prediction results
     // assert(predModel == same2Model)
-    assertDataFrameEquals(memoryPredictions, loadedPredictions)
+    assert(loadedPredictions.collect == memoryPredictions.collect)
```

Comment:
You cannot compare the equality of array elements using the `==` operator.

If the predictions are not nested arrays, you can use 
`a.sameElements(b)` where `a` and `b` are both arrays (in your case, `loadedPredictions.collect` and `memoryPredictions.collect`)

It does seem that the predictions are nested arrays, based on the Travis CI output.
In that case, `loadedPredictions.collect.deep == memoryPredictions.collect.deep` might do the trick.